src/components/ThumbsUp.jsx

```diff
@@ -1,5 +1,21 @@
-const ThumbsUp = () => {
-  return <button>Thumbs Up</button>;
+import { patchVotesInc } from "../utils/API";
+
+const ThumbsUp = ({setSingleReview, singleReview, review_id}) => {
+
+const thumbsUpHandler = () => {
+
+  setSingleReview(currReview => {
+    return {...currReview, votes: currReview.votes + 1 }
+  })
+
+  patchVotesInc(review_id).then((updatedReview) => {
+  }).catch(err => {
+    setSingleReview((currReview) => {
+      return {...currReview, votes: currReview.votes - 1}
+    }) 
+  })
+}
+  return <button className="thumbsUp" onClick={thumbsUpHandler}>Thumbs Up</button>;
 };
```

Comment:
This is great! Great use of optimistic rendering. This isn't very DRY as you could have both thumbs up and down in one component. You can pass which direction it handles on props. How would you prevent someone from clicking multiple times?